flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleMaster.java

```diff
@@ -84,6 +86,33 @@ CompletableFuture<T> registerPartitionWithProducer(
             PartitionDescriptor partitionDescriptor,
             ProducerDescriptor producerDescriptor);
 
+    /**
+     * Returns all the shuffle descriptors for the partitions in the intermediate data set with the
+     * given id.
+     *
+     * @param intermediateDataSetID The id of the intermediate data set.
+     * @return all the shuffle descriptors for the partitions in the intermediate data set. Null if
+     *     not exist.
+     */
+    default Collection<T> getClusterPartitionShuffleDescriptors(
+            IntermediateDataSetID intermediateDataSetID) {
+        return Collections.emptyList();
+    }
+
+    /**
+     * Promote the given partition to cluster partition.
+     *
+     * @param shuffleDescriptor The shuffle descriptors of the partition to promote.
+     */
+    default void promotePartition(ShuffleDescriptor shuffleDescriptor) {}
+
+    /**
+     * Remove the given partition from cluster partition.
+     *
+     * @param shuffleDescriptor The shuffle descriptors of the cluster partition to be removed.
+     */
+    default void removeClusterPartition(ShuffleDescriptor shuffleDescriptor) {}
```

Comment:
What's the difference of this method and the below one (releasePartitionExternally)?

Reply:
Thanks for the comment. You are right that we can call releasePartitionExternally to release the cluster partitions as well. 

The `removeClusterPartition` method is removed.